bindgen/codegen/mod.rs

```diff
@@ -3594,6 +3594,18 @@ impl CodeGenerator for Enum {
         let layout = enum_ty.layout(ctx);
         let variation = self.computed_enum_variation(ctx, item);
 
+        // blocklist anonymous enums based on variants.
+        if enum_ty.name().is_none() &&
+            self.is_matching_enum(
+                ctx,
+                &ctx.options().blocklisted_items,
+                item,
+            )
+        {
+            debug!("Blocklisting anonymous enum.");
+            return;
+        }
```

Comment:
Shouldn't this only block the enum if _all_ variants, and not _any_ are blocklisted? If not, can we document this behavior and why is it useful? thanks.

Reply:
The behavior of using `any` variant matches for anonymous enums is not new, and already works when allowlisting anonymous enums, so there is a (probably small) risk that changing the way matching works could break users who are relying on the current behavior for allowlisting.

Reply:
But you can individually hide variants (via annotations or callbacks), right...

I'd expect allowlisting to work with _any_ and blocklisting to work with _all_, if that makes sense... But let me know if you disagree, maybe my expectation is flawed somehow?

Reply:
No, I think that's perfectly reasonable. I changed this PR to use `all` instead. This will also block all anonymous enums without variants (See my comment on test 1025 below).